Temporarily disable connection leak tests that fail a lot

JelteF opened the pull request:
We have some leak tests that keep failing. These failures indicate problems
that should be fixed, but continuing to run them while we know they are failing
often has no benefit. This disables them temporarily, so we do not have to
rerun CI as often.

@pykello can you look at the interemediate data leak failures?

@onderkalaci can you look at the connection leak failures?

Multi intermediate data leak failures:
1. https://app.circleci.com/pipelines/github/citusdata/citus/9299/workflows/bc4515bb-4c46-44e0-bf09-b0dd485a173a/jobs/131957
2. https://app.circleci.com/pipelines/github/citusdata/citus/9290/workflows/ae5346f0-6e35-4053-b8ab-d0c690c4f18e/jobs/131804
3. https://app.circleci.com/pipelines/github/citusdata/citus/9278/workflows/f3da6bba-fc68-4b2c-b64d-ab9d0c94ec41/jobs/131595
4. https://app.circleci.com/pipelines/github/citusdata/citus/9270/workflows/072898b5-1c77-40a4-9da8-d2982ed78bc4/jobs/131476
5. https://app.circleci.com/pipelines/github/citusdata/citus/9267/workflows/bac2c4c3-5e1e-456f-9a5f-db6e69f4aec8/jobs/131422
6. https://app.circleci.com/pipelines/github/citusdata/citus/9265/workflows/d422de91-7d0f-4787-b447-5f248f3c6c5e/jobs/131377

MX connection leak failures:
1. https://app.circleci.com/pipelines/github/citusdata/citus/9296/workflows/e36d1088-662a-4f60-acec-293132632c2f/jobs/131908/steps
2. https://app.circleci.com/pipelines/github/citusdata/citus/9258/workflows/37659d82-2c5b-495e-b0e7-905811e30444/jobs/131299

Failure connection leak failures:
1. https://app.circleci.com/pipelines/github/citusdata/citus/9297/workflows/c0ebc326-8c93-468f-8b70-f470bd492fb9/jobs/131920
2. https://app.circleci.com/pipelines/github/citusdata/citus/9283/workflows/9af154d0-ff96-4c5d-ae19-81faae1e0c18/jobs/131668

SaitTalhaNisanci commented:
One problem could be that we won't know if we are introducing new leaks when they are disabled

marcocitus commented:
Could intermediate result leaks have increased after https://github.com/citusdata/citus/pull/3893 ?

onderkalaci commented:
@JelteF to clarify, there are two tests in connection leak test. One is crucial, where we keep track of the connection counts on the coordinator. I think we should not disable it. 

The other, which fails randomly, is like a double check that the worker has the number of connections open as the coordinator has opened. However, it turns out that sometimes some external backends show up.

To sum up, I think instead of completely disabling it, we should disable this part: https://github.com/citusdata/citus/blob/master/src/test/regress/sql/ensure_no_shared_connection_leak.sql#L68-L75

(And, I think commenting out that part is better, as I'll change it to SELECT * FROM ... to see which extra backends we include in the output and potentially filter them if they are not in our interest)

JelteF commented:
@marcocitus 
> Could intermediate result leaks have increased after #3893 ? I don't recall seeing this many...

Good question. I'm investigating now if it increased significantly or not. It was definitely happening before merging that too though:
1. https://app.circleci.com/pipelines/github/citusdata/citus/9126/workflows/2f7bb931-8bbb-488d-805b-accb08a3a627/jobs/129292
2. https://app.circleci.com/pipelines/github/citusdata/citus/9113/workflows/870a835b-1f3c-4465-8495-7e356eddcd2a/jobs/129081
3. https://app.circleci.com/pipelines/github/citusdata/citus/9103/workflows/c3fc6c41-4cbb-4207-9203-5e833a9d87ae/jobs/128913
4. https://app.circleci.com/pipelines/github/citusdata/citus/9067/workflows/1b28c228-75d6-46b7-b334-6ca419334869/jobs/128368

@onderkalaci I updated my change to only comment out the failing test.

JelteF commented:
I have split this PR in two, I removed the intermediate data leak test changes from this PR and added that to #3914. Since the connection leak related changes were approved by @onderkalaci I'll merge this PR now.
